cdc/processor/sinkmanager/redo_cache.go

```diff
@@ -135,10 +136,17 @@ func (e *eventAppender) pop(lowerBound, upperBound engine.Position) (res popResu
 	defer e.mu.Unlock()
 
 	if lowerBound.Compare(e.lowerBound) < 0 {
+		// NOTE: the caller will fetch events [lowerBound, res.boundary) from engine.
 		res.success = false
 		res.boundary = e.lowerBound
 		return
 	}
```

Comment:
Not sure why it can return here, could it happen that 

lowerBound < e.lowerBound < e.upperBound <= upperBound
or 
lowerBound < e.lowerBound < upperBound <= e.upperBound

Reply:
If `lowerBound < e.lowerBound` it means the cache doesn't contains all required events.
For example we want to fetch `1,2,3,4,5` but there are only `3,4,5` in cache. So we need to fetch `1,2` from engine.